src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -21,27 +21,103 @@ public function __construct(Client $guzzleClient)
     /**
      * @throws GuzzleException
      */
-    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = []): ResponseInterface
+    protected function callApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
     {
-        $request = new Request(
-            'GET',
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              [
+                'Authorization' => sprintf('Bearer %s', $bearer),
+                'Accept' => 'application/json',
+              ],
+              json_encode($bodyParams)
+          );
+        } else {
+            $request = new Request(
+              'GET',
+              sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+              ['Authorization' => sprintf('Bearer %s', $bearer)]
+          );
+        }
+
+        return $this->guzzleClient->send($request);
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    protected function deleteApi(string $uriEndpoint, string $bearer, array $queryParamsMap = [], array $bodyParams = []): ResponseInterface
+    {
+        if (count($bodyParams) > 0) {
+            $request = new Request(
+            'DELETE',
+            sprintf('%s%s', $uriEndpoint, $this->generateQueryParams($queryParamsMap)),
+            [
+              'Authorization' => sprintf('Bearer %s', $bearer),
+              'Accept' => 'application/json',
+            ],
+            json_encode($bodyParams)
+        );
```

Comment:
Indentation isn't correct here, but if you extract out a private method as suggested in another comment, it can be fixed there.